src/lib/components/SiteFooter/index.jsx

```diff
@@ -2,16 +2,33 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
 import { Hyperlink } from '@edx/paragon';
+
+import { faLanguage } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
 import messages from './SiteFooter.messages';
 
 const EVENT_NAMES = {
   FOOTER_LINK: 'edx.bi.footer.link',
 };
 
+
 class SiteFooter extends React.Component {
   constructor(props) {
     super(props);
     this.externalLinkClickHandler = this.externalLinkClickHandler.bind(this);
+    this.setLanguageSelect = this.setLanguageSelect.bind(this);
+    this.applyLanguageSelection = this.applyLanguageSelection.bind(this);
+  }
+
+  setLanguageSelect(select) {
+    this.languageSelect = select;
+  }
+
+  applyLanguageSelection(event) {
+    event.preventDefault();
+    const { languageForm: { onLanguageSelected } } = this.props;
+    onLanguageSelected(this.languageSelect.value);
```

Comment:
I think it may be better to use a form submission instead of a ref. Roughly:

In `applyLanguageSelection`
```
applyLanguageSelection(event) {
  const formData = new FormData(event.target);
  const languageValue = formData.get('language');
  // rest of the stuff
}
```

And in render:
```
<form onSubmit="this.applyLanguageSelection">
   <select name="language">
      <options/>
   </select>
   <button type="submit">Submit</button>
</form>
```